Author: Steve Langasek <steve.langasek@ubuntu.com>
Description: correct mismatched use of EFIAPI in efi shell code
 The EFIAPI define is not a no-op in GCC; it needs to be used consistently
 for both the prototype and the definition.

Index: edk2-0~20121205.edae8d2d/Shell/shellenv/shelle.h
===================================================================
--- edk2-0~20121205.edae8d2d.orig/Shell/shellenv/shelle.h
+++ edk2-0~20121205.edae8d2d/Shell/shellenv/shelle.h
@@ -1668,6 +1668,7 @@
   );
 
 EFI_STATUS
+EFIAPI
 DumpBlockDev (
   IN EFI_HANDLE               ImageHandle,
   IN EFI_SYSTEM_TABLE         *SystemTable
Index: edk2-0~20121205.edae8d2d/Shell/shellenv/cmddisp.c
===================================================================
--- edk2-0~20121205.edae8d2d.orig/Shell/shellenv/cmddisp.c
+++ edk2-0~20121205.edae8d2d/Shell/shellenv/cmddisp.c
@@ -269,12 +269,14 @@
   );
 
 EFI_STATUS
+EFIAPI
 GuidMain (
   IN EFI_HANDLE               ImageHandle,
   IN EFI_SYSTEM_TABLE         *SystemTable
   );
 
 EFI_STATUS
+EFIAPI
 OpeninfoMain (
   IN EFI_HANDLE         ImageHandle,
   IN EFI_SYSTEM_TABLE   *SystemTable
Index: edk2-0~20121205.edae8d2d/Shell/shellenv/echo.c
===================================================================
--- edk2-0~20121205.edae8d2d.orig/Shell/shellenv/echo.c
+++ edk2-0~20121205.edae8d2d/Shell/shellenv/echo.c
@@ -57,6 +57,7 @@
 };
 
 EFI_STATUS
+EFIAPI
 SEnvCmdEcho (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
Index: edk2-0~20121205.edae8d2d/Shell/hexedit/libMemImage.c
===================================================================
--- edk2-0~20121205.edae8d2d.orig/Shell/hexedit/libMemImage.c
+++ edk2-0~20121205.edae8d2d/Shell/hexedit/libMemImage.c
@@ -43,6 +43,7 @@
 };
 
 EFI_STATUS
+EFIAPI
 DummyMemRead (
   IN EFI_PCI_ROOT_BRIDGE_IO_PROTOCOL              * This,
   IN     EFI_PCI_ROOT_BRIDGE_IO_PROTOCOL_WIDTH    Width,
@@ -52,6 +53,7 @@
   );
 
 EFI_STATUS
+EFIAPI
 DummyMemWrite (
   IN EFI_PCI_ROOT_BRIDGE_IO_PROTOCOL              * This,
   IN     EFI_PCI_ROOT_BRIDGE_IO_PROTOCOL_WIDTH    Width,
@@ -392,6 +394,7 @@
 }
 
 EFI_STATUS
+EFIAPI
 DummyMemRead (
   IN EFI_PCI_ROOT_BRIDGE_IO_PROTOCOL              * This,
   IN     EFI_PCI_ROOT_BRIDGE_IO_PROTOCOL_WIDTH    Width,
@@ -404,6 +407,7 @@
 }
 
 EFI_STATUS
+EFIAPI
 DummyMemWrite (
   IN EFI_PCI_ROOT_BRIDGE_IO_PROTOCOL              * This,
   IN     EFI_PCI_ROOT_BRIDGE_IO_PROTOCOL_WIDTH    Width,
